tests/interfaces/protocols/handlers/protocols-query.spec.ts

```diff
@@ -93,6 +100,88 @@ describe('ProtocolsQueryHandler.handle()', () => {
       expect(reply2.entries?.length).to.equal(3); // expecting all 3 entries written above match the query
     });
 
+    it('should return protocols with normalized URIs matching the query', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+
+      // setting up a stub method resolver
+      TestStubGenerator.stubDidResolver(didResolver, [alice]);
+
+      // configure several protocols, all matching `example.com`
+      const protocolDefinition = emailProtocolDefinition;
+      const protocol1 = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com',
+        protocolDefinition
+      });
+      const protocolTrailingSlash = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com/',
+        protocolDefinition
+      });
+      const protocolParams = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com/?foo=bar',
+        protocolDefinition
+      });
+      const protocolCapitalized = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'EXAMPLE.Com',
+        protocolDefinition
+      });
+
+      // Sort messages into lexicographic order. We only allow protocol overwrites if the new message CID
+      // has higher lexicographic value.
+      const equivalentProtocols = [protocol1, protocolTrailingSlash, protocolParams, protocolCapitalized];
+      let messageDataWithCid: (GenerateProtocolsConfigureOutput & { cid: string })[] = [];
+      for (const messageData of equivalentProtocols) {
+        const cid = await Message.getCid(messageData.message);
+        messageDataWithCid.push({ cid, ...messageData });
+      }
+
+      messageDataWithCid = messageDataWithCid.sort((messageDataA, messageDataB) => {
+        return lexicographicalCompare(messageDataA.cid, messageDataB.cid);
+      });
+
+      // Configure protocols. Each one overwrites the previous
+      for (const protocol of messageDataWithCid) {
+        const configureReply = await dwn.processMessage(alice.did, protocol.message, protocol.dataStream);
+        expect(configureReply.status.code).to.equal(202);
+      }
+
+      // configure several more protocols, none of which match `example.com`
+      const protocolSlashFoo = await TestDataGenerator.generateProtocolsConfigure({ requester: alice, protocol: 'example.com/foo' });
+      const protocolSubdomainFoo = await TestDataGenerator.generateProtocolsConfigure({ requester: alice, protocol: 'foo.example.Com' });
```

Comment:
is the `Com` a typo or intentional?  if the latter, can you explain why?

Reply:
typo. fixed